### PR TITLE
Async islands

### PR DESCRIPTION
**Pull request by rgarnov**
<cut/>

```
force_builders=ARMv7,Custom
build_image:Custom=centos:7
buildworker:Custom=linux-1
build_gapi_standalone:Custom=ade-0.1.1f

build_gapi_standalone:Linux x64=ade-0.1.1f
build_gapi_standalone:Win64=ade-0.1.1f
build_gapi_standalone:Mac=ade-0.1.1f
build_gapi_standalone:Linux x64 Debug=ade-0.1.1f

allow_multiple_commits=1
```

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under OpenCV (BSD) License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or other license that is incompatible with OpenCV
- [x] The PR is proposed to proper branch
- [x] There is reference to original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

File: modules/gapi/src/api/gproto.cpp [hunk 1]
```diff
@@ -264,4 +264,28 @@ std::ostream& operator<<(std::ostream& os, const cv::GMetaArg &arg)
 
     return os;
 }
+} // namespace cv
+
+// TODO: This function requires thorough testing
```
Comment (dmatveev) [inline modules/gapi/src/api/gproto.cpp:269]: Can you please add these tests?

File: modules/gapi/test/streaming/gapi_streaming_tests.cpp [hunk 2]
```diff
@@ -815,6 +815,10 @@ struct GAPI_Streaming_Unit: public ::testing::Test {
     }
 };
 
+// FIXME: (GAPI_Streaming_Types,   InputOpaque) test is missing here!
+// FIXME: (GAPI_Streaming_Types, XChangeOpaque) test is missing here!
+// FIXME: (GAPI_Streaming_Types,  OutputOpaque) test is missing here!
```
Comment (dmatveev) [inline modules/gapi/test/streaming/gapi_streaming_tests.cpp:820]: Can you please file a ticket for this?

Comment (rgarnov) [reply modules/gapi/test/streaming/gapi_streaming_tests.cpp:820]: Done

File: modules/gapi/src/api/gproto.cpp (hunk 1 above)
Comment (rgarnov) [reply modules/gapi/src/api/gproto.cpp]: Added

File: modules/gapi/test/internal/gapi_int_proto_tests.cpp [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+TYPED_TEST_CASE(ProtoPtrTest, ProtoPtrTestTypes);
+
+TYPED_TEST(ProtoPtrTest, NonZero)
+{
+    typename TestFixture::Type value;
+    const auto arg = cv::gout(value).front();
+    const auto ptr = cv::gimpl::proto::ptr(arg);
+    EXPECT_EQ(ptr, &value);
+}
```
Comment (dmatveev) [inline modules/gapi/test/internal/gapi_int_proto_tests.cpp:32]: 👍

**Comment by dmatveev** [comment]
Awesome! Thanks -

**Pull request merged**